lib/astarte/core/generators/triggers/simple_events/interface_added_event.ex

```diff
@@ -0,0 +1,51 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2025 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule Astarte.Core.Generators.Triggers.SimpleEvents.InterfaceAddedEvent do
+  @moduledoc """
+  This module provides generators for Astarte Trigger Simple Event InterfaceAddedEvent struct.
+  """
+  use ExUnitProperties
+
+  import Astarte.Generators.Utilities.ParamsGen
+
+  alias Astarte.Core.Interface
+  alias Astarte.Core.Triggers.SimpleEvents.InterfaceAddedEvent
+
+  alias Astarte.Core.Generators.Interface, as: InterfaceGenerator
+
+  @spec interface_added_event() :: StreamData.t(InterfaceAddedEvent.t())
+  @spec interface_added_event(keyword :: keyword()) :: StreamData.t(InterfaceAddedEvent.t())
+  def interface_added_event(params \\ []) do
+    params gen all %Interface{
+                     name: name,
+                     major_version: major_version,
+                     minor_version: minor_version
+                   } <- InterfaceGenerator.interface(),
+                   interface <- member_of([nil, name]),
+                   major_version <- constant(major_version),
+                   minor_version <- constant(minor_version),
```

Comment:
why not having the whole interface overridable?

```suggestion
    params gen all interface <- InterfaceGenerator.interface(),
                   interface_name <- member_of([nil, interface.name]),
                   major_version <- constant(interface.major_version),
                   minor_version <- constant(interface.minor_version),
```

Reply:
after irl discussion, we have agreed to keeping parameter name and output name 1:1